src/wp-includes/class-wp-styles.php

```diff
@@ -353,6 +343,28 @@ public function print_inline_style( $handle, $display = true ) {
 		return true;
 	}
 
+	/**
+	 * Overrides the add_data method from WP_Dependencies, to allow unsetting dependencies for conditional styles.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( ! isset( $this->registered[ $handle ] ) ) {
+			return false;
+		}
+
+		if ( 'conditional' === $key ) {
+			$this->registered[ $handle ]->deps = array();
+		}
+
+		return $this->registered[ $handle ]->add_data( $key, $value );
```

Comment:
It seems preferable to rely on the parent method rather than reimplement the method here:

```suggestion
		return parent::add_data( $handle, $key, $value ); 
```

As is done in scripts:

https://github.com/WordPress/wordpress-develop/blob/57c1af7f8e408979e11fb543f676aa929cf49c62/src/wp-includes/class-wp-scripts.php#L835